src/test/java/de/tum/in/www1/artemis/exercise/quizexercise/QuizSubmissionIntegrationTest.java

```diff
@@ -401,11 +410,24 @@ void testQuizSubmitEmptyQuizInLiveMode() throws Exception {
     @WithMockUser(username = TEST_PREFIX + "student7", roles = "USER")
     @EnumSource(QuizMode.class)
     void testQuizSubmitNoDatabaseRequests(QuizMode quizMode) throws Exception {
+        CountDownLatch lock = new CountDownLatch(1);
+
+        var topic = hazelcastInstance.getTopic(HAZELCAST_CACHED_EXERCISE_UPDATE_TOPIC);
+        topic.addMessageListener(o -> lock.countDown());
+
         Course course = courseUtilService.createCourse();
         QuizExercise quizExercise = quizExerciseUtilService.createQuiz(course, ZonedDateTime.now().minusHours(5), null, quizMode);
         quizExercise.setDuration(360);
         quizExercise.getQuizBatches().forEach(batch -> batch.setStartTime(ZonedDateTime.now().minusMinutes(5)));
-        quizExerciseService.save(quizExercise);
+        quizExercise = quizExerciseService.save(quizExercise); // <- responsible for saving the exercise into the distributed exercise cache
+        final long exerciseId = quizExercise.getId();
+
+        // Wait until the exercise update got processed
+        if (!lock.await(2000, TimeUnit.MILLISECONDS)) {
+            fail("Timed out waiting for the quiz exercise cache.");
+        }
+
+        Thread.sleep(20);
```

Comment:
Why do we need these 20 milliseconds here?

Reply:
I tried to wait until not only the server recived the message, but also until it got processed and the cache got created. In my local testing the test was quite flaky without that small wait. It's not great, but I did not find another way so far. I'm happy to hear any other ideas.

Reply:
Could you maybe add a short comment why the `Thread.sleep` is needed?